repos/system_upgrade/el8toel9/actors/checkdeprecatedrpmsignature/libraries/checkdeprecatedrpmsignature.py

```diff
@@ -59,6 +60,19 @@ def _is_sha1_allowed(current_policy):
 def process():
     # TODO(pstodulk): add link to the official announce of the change in crypto policies
     bad_rpms = _get_rpms_with_sha1_sig()
+
+    # Alibaba RHUI packages (on rhel8) have to be whitelisted because it was signed with SHA1
+    upg_path = rhui.get_upg_path()
+    whitelisted_cloud_flavours = (
+        'alibaba',
+    )
+    whitelisted_cloud_pkgs = {
+        rhui.RHUI_CLOUD_MAP[upg_path].get(flavour, {}).get('src_pkg') for flavour in whitelisted_cloud_flavours
+    }
+    for bad_rpm in bad_rpms:
+        if bad_rpm.name in whitelisted_cloud_pkgs:
+            bad_rpms.remove(bad_rpm)
+
```

Comment:
this is not how it works. such packages can break the rpm transaction due to the old signature, so whitelist is not an option. it's the same as for google cloud. there are 2 options only:
* alibaba update the signature using SHA256
* user manually switch system wide crypto policies to legacy mode or DEFAULT:SHA1

Reply:
1 we cannot update   signature using SHA256 for rhel 8 repo rpm,  it will cause the upgrade to fail for rhel 7 to 8
2 this  old signature repo rpm will not  break the rpm transaction, it will be removed on upgrade, (added to remove list in system_upgrade/common/actors/cloud/checkrhui/actor.py), we test the rpm transaction is ok and upgrade successful

Reply:
@jinkkkang the problem is, that RHEL 9 rpm does not manipulate with RPM packages with SHA1 signature. if you try to install, update, or remove rpm with SHA1, openssl will not allow the use of SHA1 algorithm and rpm than fail to do anything with the rpm if the key is installed. that why I am saying that the crypto policies needs to be updated otherwise.

Why the signing of packages with SHA256 should block the upgrade 7 -> 8? The only reason I can see here is that the gpg key (in the leapp-rhui package?) needs to be updated also and than there is no reason why the upgrade should not work. All RHEL packages since RHEL 7 are signed with RSA/SHA256, so there is no technical blocker for upgrades.

Reply:
we will update  the signature of rhui repo packages on RHEL 8 using SHA256,  we will also update the leapp-rhui-alibaba package for rhel 7 upgrade  to add the sha256 gpgkey to support upgrade (rhel 7 upgrade to 8 will failed without this),   I will update here if there is any new progress, thanks

Reply:
@pirat89  hello, we have updated  the signature of  rhui repo rpm for rhel 8   from SHA1 to SHA256, we also removed the section in the code that added a whitelist to rhel 8, please help review

Reply:
Hi @jinkkkang, regarding the repomapping, we will do the change on our side later as the last time (please keep the current changes in this PR as they are in this file for now, we will update them directly manually later - we need to regenerate the file using our scripts to ensure the mapping does not dissapear in future). It seems good otherwise. I am keeping the status of required changes until we deliver the updated repomap json file (Dec / first week in Jan). Please send us the *.repo file from /etc/yum.repos.d/ for RHEL 9 on alibaba cloud (e.g via our mail communication) as the last time.

Reply:
@jinkkkang changes in repomap.json included in PR #1165 . Once it's merged, this one needs to be rebased to cover up-to-date changes and can be merged then as well. I will do it right after the change.